collaboration/sig-core/proposals/extending-kyma-cli.md

```diff
@@ -0,0 +1,476 @@
+
+`kyma function bind <function-name> --binding-instaces [<binding-instance-name>,...] -n <namespace>`
+`kyma function unbind <function-name> --binding-instaces [<binding-instance-name>,...] -n <namespace>`
+
+`kyma function bind <function-name> --binding-usage [<binding-usage-name>,...] -n <namespace>`
+`kyma function unbind <function-name> --binding-usage [<binding-usage-name>,...] -n <namespace>`
+
+*Bind to events*
+
+`kyma function bind <function-name> --events [<event-name>,..] -n <namespace>`
+`kyma function bind <function-name> --events [<event-name>,..] -n <namespace>`
+
+*Bind to environment variables*
+
+`kyma function bind <function-name> --env [<foo=bar>,..] -n <namespace>`
+`kyma function unbind <function-name> --env [<foo=bar>,..] -n <namespace>`
+
+*List all functions*
+
+`kyma function list -n <namespace> [--labels foo=bar]`
+
+> A detailed table view with function name, runtime, age, replicas and state
+
+*Runtimes*
+
+`kyma function get runtimes`
+
+*Logs*
+
+`kyma function logs <function-name> -n <namespace>`
+
+*function Status*
+
+`kyma function status <function-name> -n <namespace>`
+
+Should display status
+
+```bash
+  Name: <function-name>
+  Runtime: <my-runtime>
+  State: Running/Error/Deploying/...
+  Replicas: 1
+  Min-replicas: 1
+  Max-replicas: 2
+  API: <url>
+    Secure: yes/no
+    Actions: GET, POST
+  Bindings:
+    Events: <list of events>
+    Servicebindings: <list of service bindings>
+  labels: foo:bar
+```
+
+`kyma function show <function-name> events`
+
+`kyma function show <function-name> apis`
+
+`kyma function show <function-name> service-binding-usages`
+
+`kyma function show <function-name> labels`
+
+`kyma function show <function-name> replicas`
+
+#### Commands involved approach 1
+
+```bash
+ # Initialze project
+ kyma function init <function-name>
+  > write the desired source code and update config.yaml as per requirements`
+
+ # After testing apply to the kyma cluster.
+ # update config.yaml as per requirements like events, service bindings, env variables and api rule
+ kyma function apply -p path_to_function
+
+ # Update the function code
+ kyma function apply -p path_to_function
+```
+
+#### Commands involved approach 2
+
+ ```bash
+ # Initialze project
+ kyma function init <function-name>
+
+# After testing apply to the kyma cluster
+ kyma function apply -p path_to_function
+
+# Expose function via api rule
+ kyma function expose <function-name> --host <host_name> [--secure] [--actions <GET/POST>] -n <namespace>
+
+# Bind to events
+kyma function bind <function-name> --events [<event-name>,..] -n <namespace>
+
+# Bind to servicebinding
+kyma function bind <function-name> --binding-instaces [<binding-instance-name>,...] -n <namespace>
+
+# Bind to environment variables
+kyma function bind <function-name> --env [<foo=bar>,..] -n <namespace>
+
+# Change some code
+ kyma function update foo --src-path path_to_foo --dependecies path_to_dependecies --runtime foo-runtime -n <namespace>
```

Comment:
It should be another "apply" command, shouldn't it?

Reply:
fixed.